build-all.py

```diff
@@ -0,0 +1,216 @@
+    # print(to_make)
+
+    lv = open("build_versions.json")
+    buildver = lv.read()
+    min_tags = json.loads(buildver)
+    lv.close()
+    # min_tags look like this {'python': ['v0.9'], 'matlab': ['v3.2']}
+
+    # create full version-menu listing using the built folders from above
+    f = open(path.join('datajoint_theme', 'version-menu.html'), 'w+')
+    for lang in min_tags:
+
+        for ver in min_tags[lang]:
+        
+            f.write('<li class="version-menu"><a href="/' + lang + "/" + ver + '">' + lang + "-" + ver + '</a></li>\n')
+        
+    f.close()
+       
+    # copy over the full version-menu listing to datajoint_theme FIRST, 
+    # then build individual folders, and copy to full_site folder 
+
+    for folder in to_make:
+        shutil.copy2(path.join('datajoint_theme', 'version-menu.html'), path.join(folder, "datajoint_theme", "version-menu.html"))
+        if platform.system() == "Windows":
+            subprocess.Popen(["sphinx-build", ".", "..\_build\html"], cwd=path.join(folder, "contents")).wait() # builds html by default
+            subprocess.Popen(["sphinx-build", "-b", "latex", ".", "..\_build\latex"], cwd=path.join(folder, "contents")).wait()
+            if path.exists(path.join(folder, "site")):
+                shutil.rmtree(path.join(folder, "site"))
+            os.makedirs(path.join(folder, "site"))
+            copy_contents(path.join(folder, "_build", "html"), path.join(folder, "site"))
+        else:
+            subprocess.Popen(["make", "site"], cwd=folder).wait()
+
+        # making pdf out of the latex directory only if pdflatex runs
+        try:
+            subprocess.Popen(["pdflatex", "DataJointDocs.tex"], cwd=path.join(folder, '_build', 'latex')).wait()
+            subprocess.Popen(["pdflatex", "DataJointDocs.tex"], cwd=path.join(folder, '_build', 'latex')).wait()
+        except:
+            print("Latex environment not set up - no pdf will be generated")
+
+        lang_version = folder.split(os.sep)[1] # 'matlab-v3.2.2'
+        version = lang_version.split("-")[1].split(".")[:-1] #['v3', '2']
+        abbr_ver = ".".join(version)  #v3.2
+        abbr_lang_ver = ".".join(lang_version.split(".")[:-1])
+        shutil.copytree(path.join(folder, "site"), path.join('full_site', lang_version.split("-")[0], abbr_ver))
+
+        if path.exists(path.join(folder, '_build', 'latex', 'DataJointDocs.pdf')):
+            os.rename(path.join(folder, '_build', 'latex', 'DataJointDocs.pdf'), path.join(folder, '_build', 'latex', 'DataJointDocs_' + abbr_lang_ver + '.pdf'))
+            shutil.copy2(path.join(folder, '_build', 'latex', 'DataJointDocs_' + abbr_lang_ver + '.pdf'), path.join('full_site', lang_version.split("-")[0], abbr_ver))
+
+    for lang in ["matlab", "python"]:
+        ver_list=[]
+        for to_sort in glob.glob(path.join('full_site', lang, '**')):
+            ver_list.append(float((path.basename(to_sort).split("v")[1])))
```

Comment:
Instead of this
`'v1.2'.split('v') => ['', '1.2']`
do this
`'v1.2'.strip('v') = > '1.2'`

`'v3.5.3v'.strip('v') => '3.5.3'`
`'v3.5.3v'.lstrip('v') => '3.5.3v'`